src/AI/CapPhaseAnalyzer.java

```diff
@@ -0,0 +1,454 @@
+      startingFactories.add(dest);
+    }
+
+    buildBaseCapChains(map, viewer, rightfulProps, startingFactories);
+
+    // Add our factory chains in at the start of each list
+    for(ArrayList<CapStop> chain : factoryCapChains)
+    {
+      XYCoord start = chain.get(0).coord;
+      capChains.get(start).add(0, chain);
+    }
+  }
+
+  public void buildBaseCapChains(GameMap map, Army viewer, ArrayList<XYCoord> rightfulProps, ArrayList<XYCoord> startingFactories)
+  {
+    ArrayList<XYCoord> remainingFactories = new ArrayList<>(startingFactories);
+    // Build initial bits of capChains
+    for( XYCoord start : remainingFactories )
+    {
+      ArrayList<ArrayList<CapStop>> chainList = new ArrayList<>();
+      capChains.put(start, chainList);
+    }
+
+    boolean madeProgress = true;
+    final Unit inf = new Unit(viewer.cos[0], viewer.cos[0].getUnitModel(UnitModel.TROOP));
+    if( startingFactories.size() > 0 )
+    {
+      inf.x = startingFactories.get(0).xCoord;
+      inf.y = startingFactories.get(0).yCoord;
+    }
+    final int infMove = inf.getMovePower(map);
+
+    // Find the next stop or iterate extraTurns on all cap chains
+    while (madeProgress && !rightfulProps.isEmpty())
+    {
+      // Create new cap chains
+      for( XYCoord start : remainingFactories )
+      {
+        ArrayList<CapStop> chain = new ArrayList<>();
+        CapStop build = new CapStop(start);
+        chain.add(build);
+        capChains.get(start).add(0, chain);
+      }
+
+      madeProgress = false;
+      for( ArrayList<ArrayList<CapStop>> chainList : capChains.values() )
+        for( ArrayList<CapStop> chain : chainList )
+        {
+          if( rightfulProps.isEmpty() )
+            break;
+
+          CapStop last = chain.get(chain.size() - 1);
+          if( last.extraTurns >= LOOKAHEAD_TURNS )
+          {
+            if (chain.size() == 1)
+              remainingFactories.remove(last.coord);
+            break;
+          }
+
+          XYCoord start = last.coord;
+
+          inf.x = start.xCoord;
+          inf.y = start.yCoord;
+          Utils.sortLocationsByTravelTime(inf, rightfulProps, map);
+          XYCoord dest = rightfulProps.get(0);
+
+          final GamePath infPath = findFeasiblePath(inf, dest, map);
+          if( null == infPath || infPath.getPathLength() < 1 )
+          {
+            last.extraTurns = LOOKAHEAD_TURNS+1;
+            continue; // Can't reach
+          }
+          madeProgress = true; // We have somewhere we can still get to
+
+          final int distance = infPath.getFuelCost(inf, map);
+          final int currentTotalMove = (last.extraTurns + 1) * infMove;
+
+          if( distance <= currentTotalMove )
+          {
+            rightfulProps.remove(dest);
+            CapStop cap = new CapStop(dest);
+            chain.add(cap);
+          }
+          else
+            last.extraTurns++;
```

Comment:
Why do this iteratively instead of a single pass with `extraTurns == LOOKAHEAD_TURNS`?

Reply:
Because it's greedy.
If a city is within 2 turns of Factory B, and 3 turns of Factory A, then Factory A gets it under your scheme.

It *would* be pretty cool to find all "cap-chainable" properties and check all permutations for what's optimal, but I've spent a lot of time on this and I want to move on for now, lol

Reply:
Why would Factory A get it under that scheme? You could still see that A is farther away.
And yeah, not trying to get you to rewrite everything, just want to understand how it works.

Reply:
>start
>initialize cap chain for each factory
>look at factory A, grab city 3 turns away
>look at factory B. Don't grab the city, because it's already claimed

The current scheme is a (very) low-fidelity simulation of a cap phase - each "inf wave" travels out from its factory until it hits a specific city, then it expands out from that point.
...I just used the wave/particle duality of light as a metaphor for my code. I don't know how to feel about that.

Reply:
Well, it made sense. You may have just figured out light. Better submit that comment to a sci-mag somewhere and claim your Nobel.

Reply:
It could, yes. The iterative approach is simpler, though.
If I wanted to be really "correct" about all this, I'd build a(n a)cyclic graph of all the properties of interest and figure out what permutation of chains produces the best results.
That *might* be needed to properly handle bikes, since you don't know *a priori* where you ought to be building them to maximize funds.